redisvl/extensions/cache/llm/langcache_api.py

```diff
@@ -0,0 +1,614 @@
+import json
+from typing import Any, Dict, List, Optional, Union
+
+from langcache import LangCache as LangCacheSDK
+from langcache.models import CacheEntryScope, CacheEntryScopeTypedDict
+
+from redisvl.extensions.cache.llm.base import BaseLLMCache
+from redisvl.query.filter import FilterExpression
+from redisvl.utils.utils import current_timestamp
+
+Scope = Optional[Union[CacheEntryScope, CacheEntryScopeTypedDict]]
+
+
+class LangCache(BaseLLMCache):
+    """Redis LangCache Service: API for managing a Redis LangCache"""
+
+    def __init__(
+        self,
+        redis_client=None,
+        name: str = "llmcache",
+        distance_threshold: float = 0.1,
+        ttl: Optional[int] = None,
+        redis_url: str = "redis://localhost:6379",
+        connection_kwargs: Optional[Dict[str, Any]] = None,
+        overwrite: bool = False,
+        entry_scope: Scope = None,
+        **kwargs,
+    ):
+        """Initialize a LangCache client.
+
+        Args:
+            redis_client: A Redis client instance.
+            name: Name of the cache.
+            distance_threshold: Threshold for semantic similarity (0.0 to 1.0).
+            ttl: Time-to-live for cache entries in seconds.
+            redis_url: URL for Redis connection if no client is provided.
+            connection_kwargs: Additional Redis connection parameters.
+            overwrite: Whether to overwrite an existing cache with the same name.
+            entry_scope: Optional scope for cache entries.
+        """
+        if connection_kwargs is None:
+            connection_kwargs = {}
+
+        super().__init__(
+            name=name,
+            ttl=ttl,
+            redis_client=redis_client,
+            redis_url=redis_url,
+            connection_kwargs=connection_kwargs,
+        )
+
+        self._name = name
+        self._redis_client = redis_client
+        self._redis_url = redis_url
+        self._distance_threshold = distance_threshold
+        self._ttl = ttl
+        self._cache_id = name
+        self._entry_scope = entry_scope
+        # Initialize LangCache SDK client
+        self._api = LangCacheSDK(server_url=redis_url, client=redis_client)
+
+        # Create cache if it doesn't exist or if overwrite is True
+        try:
+            existing_cache = self._api.cache.get(cache_id=self._cache_id)
+            if not existing_cache and overwrite:
```

Comment:
what if there is not an existing cache and overwrite is false? this would init without anything behind it currently right?